utils.sh

```diff
@@ -2,6 +2,22 @@
 
 set -o pipefail
 
+function retry_with_timeout() {
+  retries=$1
+  timeout_duration=$2
+  command=${@:3}
+
+  for i in $(seq $retries); do
+    exit_code=0
+    timeout $timeout_duration bash -c "$command" || exit_code=$?
```

Comment:
you should quote function parameters:
```suggestion
    timeout "$timeout_duration" bash -c "$command" || exit_code=$?
```